spec/unit/rest/Twilio.spec.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+const nock = require('nock');
+var url = require('url');  /* jshint ignore:line */
+
+describe('client', () => {
+  var client;
+
+  beforeEach(() => {
+    var twilio = require('../../../lib');
+    client = new twilio('ACXXXXXXXX', 'test-password');
+    nock('https://api.twilio.com')
+      .get('/')
+      .reply(200, 'test response');
+  });
+
+  describe('setting the region', () => {
+    it('should use the default region if only edge is defined', () => {
+      client.edge = 'edge';
+      client.messages.create({
+        body: 'Hello from Node',
+        to: '+XXXXXXXXXX',
+        from: '+XXXXXXXXXX',
+      });
+      var uri = new url.URL(client._httpClient.lastRequest.url);
```

Comment:
Don't use `lastRequest` since the details of the underlying HTTP client could change over time. Better to use nock to mock the specific request you're expecting. See my example here: https://github.com/sendgrid/sendgrid-nodejs/pull/1093/files#diff-0a22ccfa1c28c602273fbc87ebe73ac2R41-R52